addons/purchase/models/purchase.py

```diff
@@ -1057,11 +1057,12 @@ def write(self, values):
             raise UserError(_("You cannot change the type of a purchase order line. Instead you should delete the current line and create a new line of the proper type."))
 
         if 'product_qty' in values:
-            for line in self:
-                if line.order_id.state == 'purchase':
-                    line.order_id.message_post_with_view('purchase.track_po_line_template',
-                                                         values={'line': line, 'product_qty': values['product_qty']},
-                                                         subtype_id=self.env.ref('mail.mt_note').id)
+            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+            for line in self.filtered(lambda line: line.order_id.state == 'purchase'
```

Comment:
Even though I don't think we will have a performance issue in this specific case, it's always a good practice to not iterate over a `filtered` list in order to avoid looping through the iterable 2 times (one via the filtered, the second one via the `for` loop)

You can also use nesting inversion in case your filter argument is complex. e.g.
```python
if line.order_id.state != 'purchase':
    continue
line.order_id.message_post(...)
```
instead of 
```python
if line.order_id.state == 'purchase':
    line.order_id.message_post(...)